app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java

```diff
@@ -216,44 +218,62 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         // Set initial wrap state
         initDocState();
 
+        final Runnable after = StringUtils.makeDebounced(_hlEditor, 500, () -> {
+            checkTextChangeState();
+            updateUndoRedoIconStates();
+        });
+        _hlEditor.addTextChangedListener(TextWatcherDummy.after((s) -> after.run()));
+    }
+
+    @Override
+    public void onFragmentFirstTimeVisible() {
+
         int startPos = _appSettings.getLastEditPosition(_document.getPath(), _hlEditor.length());
+        final Bundle args = getArguments();
 
         // First start - overwrite start position if needed
-        if (savedInstanceState == null) {
-            if (isDisplayedAtMainActivity()) {
+        if (_savedInstanceState == null) {
+            if (isDisplayedAtMainActivity() || _appSettings.isEditorStartOnBotttom()) {
                 startPos = _hlEditor.length();
-            } else if (args.getInt(Document.EXTRA_FILE_LINE_NUMBER, -1) >= 0) {
+            } else if (args != null && args.getInt(Document.EXTRA_FILE_LINE_NUMBER, -1) >= 0) {
                 startPos = StringUtils.getIndexFromLineOffset(_hlEditor.getText(), args.getInt(Document.EXTRA_FILE_LINE_NUMBER), 0);
-            } else if (_appSettings.isEditorStartOnBotttom()) {
-                startPos = _hlEditor.length();
             }
         }
 
         StringUtils.setSelectionAndShow(_hlEditor, startPos);
     }
 
+    public void resume() {
+        loadDocument();
+    }
+
     @Override
     public void onResume() {
+        resume();
         super.onResume();
+    }
 
-        loadDocument();
+    @Override
+    public void onSaveInstanceState(@NonNull Bundle outState) {
+        outState.putSerializable(SAVESTATE_DOCUMENT, _document);
+        super.onSaveInstanceState(outState);
+    }
 
-        if (_document != null) {
-            _document.testCreateParent();
-            boolean permok = _shareUtil.canWriteFile(_document.getFile(), false);
-            if (!permok && !_document.getFile().isDirectory() && _shareUtil.canWriteFile(_document.getFile(), _document.getFile().isDirectory())) {
-                permok = true;
-            }
-            if (_shareUtil.isUnderStorageAccessFolder(_document.getFile()) && _shareUtil.getStorageAccessFrameworkTreeUri() == null) {
-                _shareUtil.showMountSdDialog(getActivity());
-                return;
+    public void pause() {
+        saveDocument(false);
+        if (_appSettings != null && _document != null) {
+            _appSettings.addRecentDocument(_document.getFile());
+            _appSettings.setDocumentPreviewState(_document.getPath(), _isPreviewVisible);
+            if (_hlEditor != null) {
+                _appSettings.setLastEditPosition(_document.getPath(), _hlEditor.getSelectionStart());
             }
-            _textSdWarning.setVisibility(permok ? View.GONE : View.VISIBLE);
```

Comment:
Moved the permission testing into the load and save calls

Reply:
Generally .. do you have issues with the "yellow warning"? Many people talk about getting it wrongly, but it just works fine to me.